apax/md/ase_calc.py

```diff
@@ -266,6 +269,24 @@ def calculate(self, atoms, properties=["energy"], system_changes=all_changes):
         self.results = {k: np.array(v, dtype=np.float64) for k, v in results.items()}
         self.results["energy"] = self.results["energy"].item()
 
+        if self.save_to_atoms:  # Works for both cases. True and a non-empty list.
+            if self.save_to_atoms is True:  # Works only if bool is True
+                for k, v in self.results.items():
+                    if isinstance(v, float):
+                        atoms.info[k] = v
+                    else:
+                        atoms.arrays[k] = v
+            elif isinstance(self.save_to_atoms, list):
+                try:
+                    for k in self.save_to_atoms:
+                        v = self.results[k]
+                        if isinstance(v, float):
+                            atoms.info[k] = v
+                        else:
+                            atoms.arrays[k] = v
+                except Exception:
+                    pass  # add proper error handling
+
```

Comment:
Our ASE calculator is currently pretty cluttered and I don't think additional nested conditionals/loops help in that regard.
I would also not recommend to use a try except block every iteration when it really only needs to be checked once. 

My suggestions would be to do one of the following:
1. only allow for one of the two options (bool or list)
2. hardcode that enabling the option writes uncertainties to info/arrays